lib/carin_for_blue_button_test_kit/carin_search_test.rb

```diff
@@ -178,67 +179,38 @@ def run_include_search(search_params, param_value)
         if param_value != 'ExplanationOfBenefit:*'
           paths.each do |path|
               values_found = resolve_path(resource, path)
-
-              values_found.each do |reference|
-                reference_found = find_included_resource(reference, returned_resources_all)
-
-                # If at least one reference is not found, set reference_bool to false and do not change back to true for any other found references
-                if !reference_found
-                    reference_bool = false
-                end
-              end
-
               match_found = (values_found.length > 0)
 
               break if match_found
           end
           assert match_found, "Returned resource did not match the search parameter"
-          assert reference_bool, "Returned resource did not include the _include resource parameter"
-          return
         else
-          values_found = []
-          reference_bool = true
-
           paths.each do |path|
               values_found += resolve_path(resource, path)
           end
-
-          values_found.each do |reference|
-            reference_found = find_included_resource(reference, returned_resources_all)
-              
-            # If at least one reference is not found, set reference_bool to false and do not change back to true for any other found references
-            if !reference_found
-                reference_bool = false
-            end
-          end
-
           match_found = (values_found.length >= 5)
-
           assert match_found, "Returned resource did not match the search parameter"  
-          assert reference_bool, "Returned resource did not include the _include resource parameter"
         end  
-      end          
-    end
-
+      end  
 
-    def find_included_resource(reference, returned_resources_all)
-      referenced_resource_id = reference.reference
+      referenced_resource_types = values_found.map{ |reference| reference.reference.split('/')[-2]}.uniq
+      included_resources = returned_resources_all.select{|item| referenced_resource_types.include?(item.resourceType)}
 
-      assert !referenced_resource_id.start_with?('#'), "Reference id is not in the correct format of [ResourceType]/[ResourceID]"
-  
-      referenced_resource_type = referenced_resource_id.split('/')[-2]
-
-      referenced_resources = returned_resources_all.select{|item| item.resourceType == referenced_resource_type}
+      matched_base_resources = values_found.select do |base_resource_references|
+        included_resources.any? do |referenced_resource|
+          is_reference_match?(base_resource_references.reference, referenced_resource.id)
```

Comment:
Is the "referenced_resource.id" really just the id? This value should be a local reference format, ie [resource_type]/[resource_id]